### ISSUE TITLE
Gala Crashes when Notifications are Displayed

### ISSUE THREAD

**Issue by naftulikay** [elementary/gala#1456]
### What Happened?

If <kbd>Do Not Disturb</kbd> is <kbd>Off</kbd> and notifications come in from Slack (installed via Flatpak: `com.slack.Slack`) or Signal Desktop (installed via Signal's Debian repository), **Gala often crashes**. It probably also occurs from other applications as well, such as Discord (installed via Flatpak: `com.discordapp.Discord`).

If I enable <kbd>Do Not Disturb</kbd>, I cannot reproduce this issue, so somehow the notification system is crashing Gala.

This means that if I want stability, I have to have <kbd>Do Not Disturb</kbd> on, so I can't see notifications.

### Steps to Reproduce

Install Slack, Discord, and/or Signal Messenger, and have a friend message you, creating notifications. Often, this will lead to Gala crashing. It's certainly possible that other notifications will cause it to crash. It doesn't happen every time, but often enough that I leave <kbd>Do Not Disturb</kbd> on.

### Expected Behavior

Gala should not crash when notifications arrive.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
Aug 24 13:25:53 hollow gala.desktop[62865]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x1e00003
Aug 24 13:26:20 hollow gala[62865]: clutter_actor_add_effect_with_name: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 13:26:31 hollow gala.desktop[62865]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x1e00003
Aug 24 13:26:37 hollow gala.desktop[62865]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x0 specified for 0x9800057.
Aug 24 13:26:43 hollow gala[62865]: clutter_actor_add_effect_with_name: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 13:31:44 hollow gala[62865]: clutter_actor_set_child_below_sibling: assertion 'child->priv->parent == self' failed
Aug 24 13:31:44 hollow gala[62865]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 13:31:44 hollow gala[62865]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 13:31:44 hollow gala[62865]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 13:31:45 hollow gala.desktop[62865]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x6a0006e
Aug 24 13:58:02 hollow gala[62865]: clutter_actor_destroy: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 13:58:02 hollow gala[62865]: clutter_actor_destroy: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 13:58:02 hollow gala[62865]: clutter_actor_destroy: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 13:58:02 hollow gala[62865]: clutter_actor_destroy: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 14:01:39 hollow gala.desktop[2645]: Window manager warning: Trying to re-add keybinding "switch-to-workspace-last".
Aug 24 14:01:39 hollow gala.desktop[2645]: Window manager warning: Trying to re-add keybinding "move-to-workspace-last".
Aug 24 14:01:42 hollow gala[2645]: STACK_OP_RAISE_ABOVE: sibling window 0x2800007 (io.element) not in stack
Aug 24 14:01:42 hollow gala[2645]: STACK_OP_RAISE_ABOVE: sibling window 0x2800007 not in stack
Aug 24 14:01:52 hollow gala[2645]: clutter_actor_add_effect_with_name: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 14:04:10 hollow gala[2645]: clutter_actor_set_child_below_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:04:10 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:04:10 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:04:10 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:04:11 hollow gala.desktop[2645]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x3800014
Aug 24 14:21:50 hollow gala[2645]: clutter_actor_set_child_below_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:21:50 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:21:50 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 14:21:50 hollow gala[2645]: clutter_actor_set_child_above_sibling: assertion 'child->priv->parent == self' failed
Aug 24 15:43:21 hollow gala[2645]: clutter_actor_add_effect_with_name: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 15:43:23 hollow gala[2645]: clutter_actor_add_effect_with_name: assertion 'CLUTTER_IS_ACTOR (self)' failed
Aug 24 16:00:36 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:36 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:37 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:38 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:38 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:40 hollow gala[2645]: Error in map accounting.
Aug 24 16:00:40 hollow gala[2645]: Error in map accounting.
Aug 25 11:48:50 hollow gala.desktop[2645]: Window manager warning: Tried to ping window 0x3800047 with a bad serial! Not allowed.
Aug 25 12:09:38 hollow gala.desktop[2645]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x1e00003
Aug 25 12:10:22 hollow gala.desktop[2645]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x1a00007
Aug 25 12:27:13 hollow gala.desktop[2645]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x6800007
```

### Hardware Info

Lenovo Thinkpad X1 Yoga 4th generation, connected to a Lenovo Thunderbolt USB-C dock, connected to an external monitor over DisplayPort. The laptop is closed and I'm not using the built-in display. I don't know if I have observed it crashing when not using an external monitor.

I'm on kernel 5.15.0-46-generic, it has a i7-8665U CPU, 16GiB of RAM, NVMe for storage, integrated Intel graphics 620.

**Comment by naftulikay** [comment]
Here are logs from an actual crash that just occurred:

```text
Aug 25 15:54:18 hollow gnome-session[2358]: gnome-session-binary[2358]: WARNING: Application 'gala.desktop' killed by signal 11
Aug 25 15:54:18 hollow gnome-session-binary[2358]: WARNING: Application 'gala.desktop' killed by signal 11
Aug 25 15:54:21 hollow gala.desktop[30398]: Window manager warning: Trying to re-add keybinding "switch-to-workspace-last".
Aug 25 15:54:21 hollow gala.desktop[30398]: Window manager warning: Trying to re-add keybinding "move-to-workspace-last".
Aug 25 15:54:21 hollow gala[30398]: meta_sound_player_play_from_theme: assertion 'META_IS_SOUND_PLAYER (player)' failed

```

**Comment by GranPC** [comment]
There's some more discussion on this issue in elementary/notifications#139. Still an issue in daily builds. Going to update that other issue with some info.

**Comment by lenemter** [comment]
Fixed in #1497
